lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureCloudBlobContainer.cs

```diff
@@ -25,22 +25,24 @@ namespace Org.Apache.REEF.IO.FileSystem.AzureBlob
     internal sealed class AzureCloudBlobContainer : ICloudBlobContainer
     {
         private readonly CloudBlobContainer _container;
+        private readonly BlobRequestOptions _requestOptions;
 
-        public AzureCloudBlobContainer(CloudBlobContainer container)
+        public AzureCloudBlobContainer(CloudBlobContainer container, BlobRequestOptions requestOptions)
         {
             _container = container;
+            _requestOptions = requestOptions;
         }
 
         public bool CreateIfNotExists()
         {
-            var task = _container.CreateIfNotExistsAsync();
+            var task = _container.CreateIfNotExistsAsync(_requestOptions, null);
             task.Wait();
             return task.Result;
```

Comment:
nit: feel free to clean these up if you ever have the time 
```csharp
return _container.CreateIfNotExistsAsync(_requestOptions, null).Wait().Result;
```

Reply:
Wait() return is void

---
In reply to: [194559199](https://github.com/apache/reef/pull/1470#discussion_r194559199) [](ancestors = 194559199)

Reply:
Ahhh...my bad.

---
In reply to: [194583416](https://github.com/apache/reef/pull/1470#discussion_r194583416) [](ancestors = 194583416,194559199)

Reply:
still, why do we need `.Wait()` here? I think querying the `.Result`  would wait for the task completion anyway?

Reply:
removed

---
In reply to: [197576694](https://github.com/apache/reef/pull/1470#discussion_r197576694) [](ancestors = 197576694)